src/PixelFlags.cc

```diff
@@ -57,87 +59,153 @@ class FootprintBits : public afw::detection::FootprintFunctor<MaskedImageT> {
+                                        "Source is outside usable exposure region (masked EDGE or NO_DATA)");
+    _predefinedAnyKeys["INTRP"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_interpolated",
+                                        "Interpolated pixel in the source footprint");
+    _predefinedAnyKeys["SAT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_saturated",
+                                        "Saturated pixel in the source footprint");
+    _predefinedAnyKeys["CR"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_cr",
+                                        "Cosmic ray in the source footprint");
+    _predefinedAnyKeys["BAD"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_bad",
+                                        "Bad pixel in the source footprint");
+    _predefinedAnyKeys["SUSPECT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_suspect",
+                                        "source's footprint includes suspect pixels");
+    // Flags that correspond to mask bits which occur in the center of the object
+    _predefinedCenterKeys["INTRP"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_interpolatedCenter",
+                                        "Interpolated pixel in the source center");
+    _predefinedCenterKeys["SAT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_saturatedCenter",
+                                        "Saturated pixel in the source center");
+    _predefinedCenterKeys["CR"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_crCenter",
+                                        "Cosmic ray in the source center");
+    _predefinedCenterKeys["SUSPECT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_suspectCenter",
+                                        "source's center is close to suspect pixels");
+
+    // Read in the flags passed from the configuration, and add them to the schema
+    for (std::vector<std::string>::const_iterator i = _ctrl.center.begin(); i != _ctrl.center.end(); ++i){
+        std::string tname(*i);
+        std::transform(tname.begin(), tname.end(), tname.begin(), ::tolower);
+        _centerKeys[*i] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_"+tname+"Center",
+                                        "source center is close to "+ *i + " pixels");
+    }
+    for (std::vector<std::string>::const_iterator i = _ctrl.any.begin(); i != _ctrl.any.end(); ++i){
+        std::string tname(*i);
+        std::transform(tname.begin(), tname.end(), tname.begin(), ::tolower);
+        _anyKeys[*i] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname + "_" + tname,
+                                        "source footprint includes " + *i + " pixels");
+    }
 }
+void PixelFlagsAlgorithm::updateFlags(const KeyMap * maskFlagToPixelFlag, const FootprintBits<MaskedImageT> * func,
```

Comment:
`const` comes after the type name.

As per earlier comments, this looks like it should be a free function taking references rather than a member taking pointers.